File: appbase/src/main/java/cool/dingstock/appbase/ext/ViewPagerHelper.java
package cool.dingstock.appbase.ext;

import androidx.viewpager2.widget.ViewPager2;

import net.lucode.hackware.magicindicator.MagicIndicator;

public class ViewPagerHelper {
	public static void bind(final MagicIndicator magicIndicator, ViewPager2 viewPager) {
		viewPager.registerOnPageChangeCallback(new ViewPager2.OnPageChangeCallback() {
			@Override
			public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
				magicIndicator.onPageScrolled(position, positionOffset, positionOffsetPixels);
			}
			
			@Override
			public void onPageSelected(int position) {
				magicIndicator.onPageSelected(position);
			}
			
			@Override
			public void onPageScrollStateChanged(int state) {
				magicIndicator.onPageScrollStateChanged(state);
			}
		});
	}
}

File: appbase/src/main/kotlin/cool/dingstock/appbase/uikit/widget/magicindicator/ViewPagerHelper.kt
package cool.dingstock.appbase.uikit.widget.magicindicator

import androidx.viewpager.widget.ViewPager
import androidx.viewpager.widget.ViewPager.OnPageChangeListener


object ViewPagerHelper {
    fun bind(magicIndicator: MagicIndicator, viewPager: ViewPager) {
        viewPager.addOnPageChangeListener(object : OnPageChangeListener {
            override fun onPageScrolled(
                position: Int,
                positionOffset: Float,
                positionOffsetPixels: Int
            ) {
                magicIndicator.onPageScrolled(position, positionOffset, positionOffsetPixels)
            }

            override fun onPageSelected(position: Int) {
                magicIndicator.onPageSelected(position)
            }

            override fun onPageScrollStateChanged(state: Int) {
                magicIndicator.onPageScrollStateChanged(state)
            }
        })
    }
}
